### PR TITLE
Add Configure a BSP Connection to MicroEJ Platform "FullPackaging"

### PR DESCRIPTION
**Pull request by gjadi-mej**
https://docs.microej.com/en/feature-m0018pqt-250_tuto_fullpackaging_convertion/PlatformDeveloperGuide/platformMigration.html

### CODE REVIEW

File: PlatformDeveloperGuide/platform.rst [hunk 1]
```diff
@@ -9,6 +9,7 @@ MicroEJ Platform
    platformConcepts
    platformCreation
    platformQualification
+   platformConvertFullPackaging
```
Comment (frivieremicroej) [inline PlatformDeveloperGuide/platform.rst:12]: After my review, we will ask @fpinel-microej to run again the steps

File: PlatformDeveloperGuide/platformMigration.rst [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+                MicroEJ Architecture.
+
+  - Open the ``release_pack.properties`` file:
+
+    - ``packName`` is the ``NAME`` (e.g. ``fs``)
+    - ``packVersion`` is the ``VERSION`` (e.g. ``4.0.2``)
+
+  For example, in the STM32F746G-DISCO Platform, the Legacy MicroEJ
+  Generic Pack FS ``fs.xpfp`` shall be copied and renamed to
+  ``architecture-repository/com/microej/pack/fs/4.0.2/fs-4.0.2.xpfp``.
```
[the remark this answers is not in the document]
Comment (fpinel-microej) [reply PlatformDeveloperGuide/platformMigration.rst:132]: Actually this should follow the convention for legacy packs, so this was correct (but would benefit for a small clarification ?)

File: PlatformDeveloperGuide/platformMigration.rst [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+  for the C TOOLCHAIN of the BSP
+  (e.g. ``PlatformQualificationTools/framework/platform/scripts/KEILuV5/``)
+- Configure the scripts.
+- Enable the execution of the build script: 
+
+  - Go to :guilabel:`Run` > :guilabel:`Run Configurations...`
+  - Select the launch configuration
+  - Go to :guilabel:`Configuration` > :guilabel:`Device` > :guilabel:`Deploy`
+  - Ensure :guilabel:`Execute the MicroEJ build script (build.bat) at
+    a location known by the 3rd-party BSP project.` is checked.
```
[the remark this answers is not in the document]
Comment (fpinel-microej) [reply PlatformDeveloperGuide/platformMigration.rst:318]: UPDATE : The issue came from the `KEIL_CHANGEIT_INSTALLATION_DIR` property that I set to `C:\Keil_v5\UV4` (the reasoning was that this is where the executable for Keil Uvision is), but the script only expects `C:\Keil_v5` (as indicated in the comment above the property) and adds the `\UV4` folder to the path itself.

File: PlatformDeveloperGuide/platformMigration.rst [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+    - ``architecture`` is the ``ISA`` (e.g. ``CM7``)
+    - ``toolchain`` is the ``TOOLCHAIN`` (e.g. ``CM7hardfp_ARMCC5``)
+    - ``name`` is the ``UID`` (e.g. ``flopi7A21``)
+    - ``packName`` is the ``NAME`` (e.g. ``ui``)
+    - ``packVersion`` is the ``VERSION`` (e.g. ``12.0.1``)
+
+  For example, in the STM32F746G-DISCO Platform, the MicroEJ
+  Architecture Specific Pack UI ``flopi7A21UI.xpfp`` shall be copied
+  and renamed to
+  ``architecture-repository/com/microej/architecture/CM7/CM7hardfp_ARMCC5/flopi7A21-ui-pack/12.0.1/flopi7A21-ui-pack-12.0.1.xpfp``.
```
[the remark this answers is not in the document]
Comment (gjadi-mej) [reply PlatformDeveloperGuide/platformMigration.rst:114]: Updated naming convention.  The artifact should have `-pack`, see https://repository.microej.com/architectures/com/microej/architecture/CM7/CM7hardfp_GCC48/flopi7G26-ui-pack/13.0.4/.

File: PlatformDeveloperGuide/platformMigration.rst [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+
+  - Go to :guilabel:`Window` > :guilabel:`Preferences` >
+    :guilabel:`MicroEJ` > :guilabel:`Module Manager`
+  - In :guilabel:`Module Repository` set :guilabel:`Settings File:` to
+    ``${workspace_loc:architecture-repository/ivysettings.xml}``.
+  - :guilabel:`Apply and Close`
+
+Here is the layout of the Architecture Repository for STM32F746G-DISCO.
+
+.. figure:: images/platform_architecture-repository.png
```
[the remark this answers is not in the document]
Comment (gjadi-mej) [reply PlatformDeveloperGuide/platformMigration.rst:144]: Regarding Standalone pack, I added a sentence to explain that it should not be imported.

File: PlatformDeveloperGuide/platformMigration.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (gjadi-mej) [reply PlatformDeveloperGuide/platformMigration.rst:132]: Added the naming convention for Legacy MicroEJ Generic Pack.

File: PlatformDeveloperGuide/platformMigration.rst [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+<https://github.com/MicroEJ/PlatformQualificationTools/tree/master/framework/platform/scripts>`__.
+
+On the STM32F746G-DISCO, the C TOOLCHAIN used is Keil uVision.
+
+- Create the directory pointed by ``microejscript.relative.dir``
+  (e.g. ``STM32F746GDISCO-Full-CM7_ARMCC-FreeRTOS-bsp\Projects\STM32746G-Discovery\Applications\MicroEJ\scripts``).
+- Copy the example scripts from the `Platform Qualification Tools`_
+  for the C TOOLCHAIN of the BSP
+  (e.g. ``PlatformQualificationTools/framework/platform/scripts/KEILuV5/``)
+- Configure the scripts.
```
[the remark this answers is not in the document]
Comment (gjadi-mej) [reply PlatformDeveloperGuide/platformMigration.rst:311]: Added a mention that the user should refer to the documentation of the script in comments.  Is that clear enough?

[the remark this answers is not in the document]
Comment (fpinel-microej) [reply PlatformDeveloperGuide/platformMigration.rst]: I think it should be good, the exemples given in the comments of the scripts worked fine for this tutorial so this redirection should be enough.

[the remark this answers is not in the document]
Comment (gjadi-mej) [reply PlatformDeveloperGuide/platformMigration.rst]: Cool, thanks.

**Pull request merged**